.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,93 @@
+# .github/workflows/pr-preview.yml
+name: Deploy PR previews
+
+on:
+  pull_request:
+    types:
```

Comment:
triggering the action on open/reopen/closed seems not useful here. 

Can we do something like 'on `pull_request` against master" and: 
```
on:
  push:
    tags: 
      - preview
```
To also trigger it when we add the `preview` label? (note: I am not sure if this is triggered also when a new label is added)

We should also check that the action is not run on PR if the `preview` label is not present. Otherwise this can create a lot of concurrent PRs and updates